eureka2-core/src/main/java/com/netflix/eureka2/registry/NotifyingInstanceInfoHolder.java

```diff
@@ -255,4 +274,94 @@ public void shutdown() {
             super.shutdown();
         }
     }
+
+    /**
+     * Assume access to this is synchronized.
+     * All access to the dataStore first route through the sourceMap for the authoritative source to use
+     */
+    /* visible for testing */ static class DataStore {
+        protected final Map<String, Source> sourceMap = new HashMap<>();
```

Comment:
Can we use custom comparator (by origin/name) and eliminate this data structure?